test/fail_compilation/test8006.d

```diff
@@ -0,0 +1,69 @@
+/*
+ * TEST_OUTPUT:
+---
+fail_compilation/test8006.d(48): Error: function test8006.TInt.x () is not callable using argument types (int)
```

Comment:
Agreed.  I looked into this and found that the error messages really need to be improved for all usages of properties, not just for the binary assignments in this pull request.

The error message situation is broader than the scope of this pull request.  Simple property assignments are currently implemented and produce the same error message as above.  So this pull request maintains the status quo.

The following scenarios all produce different error messages, none of which read specifically to syntax of property usage.

- Attempting to set without a setter
- Attempting to get without a getter
- Attempting to set without a setter through `alias this`
- Attempting to get without a getter through `alias this`
- Attempting to set without a setter through `static alias this`
- Attempting to get without a getter through `static alias this`
- Attempting to set a property to the wrong type
- Attempting to set a property to the wrong type through `alias this`
- Attempting to set a property to the wrong type through `static alias this`
- etc...

So, instead of improving the error messages in this pull request, I suggest accepting this pull request (maintaining the status quo) and improve the error messages for properties more comprehensively in a separate issue and pull request.